source/Octopus.Client/Repositories/Async/DeploymentProcessRepository.cs

```diff
@@ -6,19 +6,59 @@ namespace Octopus.Client.Repositories.Async
 {
     public interface IDeploymentProcessRepository : IGet<DeploymentProcessResource>, IModify<DeploymentProcessResource>
     {
+        IDeploymentProcessRepositoryBeta Beta(bool useBeta);
         Task<ReleaseTemplateResource> GetTemplate(DeploymentProcessResource deploymentProcess, ChannelResource channel);
     }
 
     class DeploymentProcessRepository : BasicRepository<DeploymentProcessResource>, IDeploymentProcessRepository
     {
+        private readonly IDeploymentProcessRepositoryBeta beta;
+
         public DeploymentProcessRepository(IOctopusAsyncRepository repository)
             : base(repository, "DeploymentProcesses")
         {
+            beta = new DeploymentProcessRepositoryBeta(repository);
+        }
+
+        public IDeploymentProcessRepositoryBeta Beta(bool useBeta)
+        {
+            if (!useBeta) throw new Exception($"You must supply true for {nameof(useBeta)} to use Beta functionality.");
```

Comment:
This is likely overkill, it's obvious to me from the property name that it's beta and I'm opting in to ☠️